fix(ci): 3 build failures on main after lgtm-ci migration

## Summary

Three workflows fail on every push to main after the lgtm-ci v0.18.1 migration (PR #939): **CI - Tests** (coverage pages publish), **Security - SBOM Generation** (grype vuln scan), and **CI - Docker** (GHCR publish). All are egress/artifact regressions introduced by the new workflow layout.

## Problem

### 1. CI - Tests — `Publish Coverage to Pages` fails

```
##[error]Unable to download artifact(s): Artifact not found for name: python-coverage
```

- **Root cause:** `reusable-test-python.yml` with `python-versions: '3.11,3.14'` uploads per-version artifacts (`python-coverage-3.11`, `python-coverage-3.14`). `reusable-test-python-publish.yml` expects a single `python-coverage` artifact. No merge step exists.
- **Failed run:** https://github.com/lgtm-hq/py-lintro/actions/runs/26370278488

### 2. Security - SBOM Generation — `Scan vulnerabilities` fails

```
unable to download listing: Get "https://grype.anchore.io/databases/v6/latest.json": connection refused
failed to load vulnerability db: database does not exist
```

- **Root cause:** Grype v6 fetches its DB from `grype.anchore.io:443`, which is missing from the SBOM egress allowlist. Harden Runner blocks it.
- **Failed run:** https://github.com/lgtm-hq/py-lintro/actions/runs/26370278431

### 3. CI - Docker — `📦 Publish to GHCR` fails

```
ERROR: Error response from daemon: Get "https://registry-1.docker.io/v2/": connection refused
```

- **Root cause:** The `publish` job uses Buildx `driver: docker-container`, which pulls `moby/buildkit:buildx-stable-1` from Docker Hub. The egress allowlist only has `ghcr.io:443` — no Docker Hub hosts (`registry-1.docker.io:443`, `auth.docker.io:443`, etc.). The sibling `docker-build` job already has them.
- **Failed run:** https://github.com/lgtm-hq/py-lintro/actions/runs/26370278421

## Proposed Solution

### Fix 1: Coverage artifact merge job (`test-ci.yml`)

Add `merge-coverage` job using `actions/upload-artifact/merge` to combine per-version `python-coverage-*` artifacts into `python-coverage` before the publish step.

### Fix 2: Grype egress host (`sbom-on-main.yml`, `publish-pypi-on-tag.yml`)

Add `grype.anchore.io:443` to both SBOM egress allowlists.

### Fix 3: Docker Hub egress on publish (`docker-ci.yml`)

Align the `publish` job's egress allowlist with `docker-build` — add Docker Hub hosts plus all build-time hosts (apt, pip, rustup, npm, crates.io, semgrep).

## References

- PR #939 — lgtm-ci migration that introduced these workflows
- Issue #938 — tracking issue for the migration
- lgtm-ci#210 — upstream gap: reusable test workflow should merge coverage artifacts

--- a/.github/workflows/README.md
+++ b/.github/workflows/README.md
@@ -3,9 +3,11 @@
 This repository uses GitHub Actions for quality gates, release automation, and
 publishing. Shared workflows are thin callers to
 [lgtm-ci](https://github.com/lgtm-hq/lgtm-ci) reusable workflows pinned at
-`22a7c110415d75843c2c216f95ed47aaeb8e26fc` (**v0.18.1**; includes
+`e42d374f1f89e0ad20d165cd611eb7732462b581` (**v0.18.2**; includes
 [lgtm-ci#206](https://github.com/lgtm-hq/lgtm-ci/pull/206) host-user docker quality and
-`lintro-tool-options` passthrough).
+`lintro-tool-options` passthrough,
+[lgtm-ci#211](https://github.com/lgtm-hq/lgtm-ci/pull/211) per-version coverage artifact
+merge).
 
 ## CI (main branch)
 

--- a/.github/workflows/codeql.yml
+++ b/.github/workflows/codeql.yml
@@ -24,7 +24,7 @@ concurrency:
 jobs:
   codeql:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-codeql.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-codeql.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       languages: python
       build-mode: none

--- a/.github/workflows/dependency-review.yml
+++ b/.github/workflows/dependency-review.yml
@@ -17,7 +17,7 @@ concurrency:
 jobs:
   dependency-review:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-dependency-review.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-dependency-review.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       fail-on-severity: high
       egress-policy: block

--- a/.github/workflows/docker-build-publish.yml
+++ b/.github/workflows/docker-build-publish.yml
@@ -23,7 +23,7 @@ jobs:
   docker-full:
     name: 🐳 Docker Full Image
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-docker.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-docker.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     permissions:
       contents: read
       packages: write
@@ -52,7 +52,7 @@ jobs:
       no-cache: ${{ startsWith(github.ref, 'refs/tags/') }}
       provenance: false
       sbom: false
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443
@@ -86,7 +86,7 @@ jobs:
     name: 🐳 Docker Base Image
     needs: docker-full
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-docker.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-docker.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     permissions:
       contents: read
       packages: write
@@ -115,7 +115,7 @@ jobs:
       no-cache: ${{ startsWith(github.ref, 'refs/tags/') }}
       provenance: false
       sbom: false
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443

--- a/.github/workflows/docker-ci.yml
+++ b/.github/workflows/docker-ci.yml
@@ -469,6 +469,25 @@ jobs:
             api.github.com:443
             ghcr.io:443
             pkg-containers.githubusercontent.com:443
+            docker.io:443
+            registry-1.docker.io:443
+            auth.docker.io:443
+            production.cloudflare.docker.com:443
+            production.cloudfront.docker.com:443
+            deb.debian.org:80
+            pypi.org:443
+            files.pythonhosted.org:443
+            static.rust-lang.org:443
+            bun.sh:443
+            astral.sh:443
+            releases.astral.sh:443
+            sh.rustup.rs:443
+            registry.npmjs.org:443
+            crates.io:443
+            static.crates.io:443
+            index.crates.io:443
+            semgrep.dev:443
+            metrics.semgrep.dev:443
       - name: Checkout
         uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6.0.2
         with:

--- a/.github/workflows/docker-ci.yml
+++ b/.github/workflows/docker-ci.yml
@@ -210,13 +210,13 @@ jobs:
   dogfooding-quality:
     needs: [docker-build, manifest-sync]
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-quality.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-quality.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     permissions:
       contents: read
       packages: read # pull lintro-image from GHCR in reusable-quality
       pull-requests: write # reusable-quality post-pr-comment step
     with:
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       job-name: '🛠️ Dogfooding Lint'
       post-pr-comment: true
       lintro-tool-options: >-
@@ -345,7 +345,7 @@ jobs:
       - name: Comment PR with security audit results
         if: github.event_name == 'pull_request'
         continue-on-error: true
-        uses: lgtm-hq/lgtm-ci/.github/actions/post-pr-comment@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+        uses: lgtm-hq/lgtm-ci/.github/actions/post-pr-comment@e42d374f1f89e0ad20d165cd611eb7732462b581
         with:
           file: security-audit-comment.txt
           marker: security-audit-report

--- a/.github/workflows/pr-auto-assign.yml
+++ b/.github/workflows/pr-auto-assign.yml
@@ -12,14 +12,14 @@ permissions: {}
 jobs:
   assign:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-pr-auto-assign.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-pr-auto-assign.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       codeowners-path: .github/CODEOWNERS
       egress-policy: block
       allowed-endpoints: >
         github.com:443
         api.github.com:443
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
     permissions:
       contents: read
       pull-requests: write

--- a/.github/workflows/pr-labeler.yml
+++ b/.github/workflows/pr-labeler.yml
@@ -12,7 +12,7 @@ permissions: {}
 jobs:
   label:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-pr-labeler.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-pr-labeler.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       configuration-path: .github/labeler.yml
       sync-labels: false

--- a/.github/workflows/publish-pypi-on-tag.yml
+++ b/.github/workflows/publish-pypi-on-tag.yml
@@ -28,10 +28,10 @@ jobs:
   quality:
     if: ${{ !startsWith(github.ref_name, 'actions-v') }}
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-quality.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-quality.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       post-pr-comment: false
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443
@@ -46,7 +46,7 @@ jobs:
     name: Generate and verify SBOM
     if: ${{ !startsWith(github.ref_name, 'actions-v') }}
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-sbom.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-sbom.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       target: .
       target-type: dir
@@ -71,7 +71,7 @@ jobs:
         timestamp.sigstore.dev:443
         tuf-repo-cdn.sigstore.dev:443
         sigstore-tuf-root.storage.googleapis.com:443
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
     permissions:
       contents: read
       security-events: write
@@ -111,7 +111,7 @@ jobs:
         with:
           repository: lgtm-hq/lgtm-ci
           path: .lgtm-ci-tooling
-          ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+          ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
           sparse-checkout: scripts/ci/
           sparse-checkout-cone-mode: true
           persist-credentials: false
@@ -170,7 +170,7 @@ jobs:
         with:
           repository: lgtm-hq/lgtm-ci
           path: .lgtm-ci-tooling
-          ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+          ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
           sparse-checkout: scripts/ci/
           sparse-checkout-cone-mode: true
           persist-credentials: false

--- a/.github/workflows/publish-pypi-on-tag.yml
+++ b/.github/workflows/publish-pypi-on-tag.yml
@@ -64,6 +64,7 @@ jobs:
         anchore.io:443
         get.anchore.io:443
         toolbox-data.anchore.io:443
+        grype.anchore.io:443
         pipelines.actions.githubusercontent.com:443
         fulcio.sigstore.dev:443
         rekor.sigstore.dev:443

--- a/.github/workflows/publish-testpypi.yml
+++ b/.github/workflows/publish-testpypi.yml
@@ -11,12 +11,12 @@ permissions: {}
 jobs:
   publish:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-publish-pypi.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-publish-pypi.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       test-pypi: true
       update-homebrew: false
       python-version: '3.14'
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443

--- a/.github/workflows/release-auto-tag.yml
+++ b/.github/workflows/release-auto-tag.yml
@@ -17,10 +17,10 @@ concurrency:
 jobs:
   auto-tag:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-release-auto-tag.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-release-auto-tag.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       create-release: false
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443

--- a/.github/workflows/release-version-pr.yml
+++ b/.github/workflows/release-version-pr.yml
@@ -17,12 +17,12 @@ concurrency:
 jobs:
   version-pr:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-release-version-pr.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-release-version-pr.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       ecosystems: python
       auto-merge: true
       max-bump: minor
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443

--- a/.github/workflows/sbom-on-main.yml
+++ b/.github/workflows/sbom-on-main.yml
@@ -35,6 +35,7 @@ jobs:
         anchore.io:443
         get.anchore.io:443
         toolbox-data.anchore.io:443
+        grype.anchore.io:443
         pipelines.actions.githubusercontent.com:443
         fulcio.sigstore.dev:443
         rekor.sigstore.dev:443

--- a/.github/workflows/sbom-on-main.yml
+++ b/.github/workflows/sbom-on-main.yml
@@ -17,7 +17,7 @@ concurrency:
 jobs:
   sbom:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-sbom.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-sbom.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       target: .
       target-type: dir
@@ -42,7 +42,7 @@ jobs:
         timestamp.sigstore.dev:443
         tuf-repo-cdn.sigstore.dev:443
         sigstore-tuf-root.storage.googleapis.com:443
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
     permissions:
       contents: read
       security-events: write

--- a/.github/workflows/scorecards.yml
+++ b/.github/workflows/scorecards.yml
@@ -17,7 +17,7 @@ concurrency:
 jobs:
   scorecards:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-scorecards.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-scorecards.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       egress-policy: block
       allowed-endpoints: >

--- a/.github/workflows/semantic-pr-title.yml
+++ b/.github/workflows/semantic-pr-title.yml
@@ -17,7 +17,7 @@ concurrency:
 jobs:
   semantic-title:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-semantic-pr-title.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-semantic-pr-title.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       egress-policy: block
       allowed-endpoints: >

--- a/.github/workflows/test-ci.yml
+++ b/.github/workflows/test-ci.yml
@@ -93,7 +93,6 @@ jobs:
         with:
           name: python-coverage
           pattern: python-coverage-*
-          delete-merged: true
           retention-days: 7
 
   publish-coverage:

--- a/.github/workflows/test-ci.yml
+++ b/.github/workflows/test-ci.yml
@@ -21,7 +21,7 @@ concurrency:
 jobs:
   test:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-test-python.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-test-python.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       post-pr-comment: true
       python-versions: '3.11,3.14'
@@ -35,7 +35,7 @@ jobs:
       draft-pr-skip: true
       job-name: Python Compatibility
       runner-image: ubuntu-24.04
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443
@@ -76,40 +76,21 @@ jobs:
             exit 1
           fi
 
-  merge-coverage:
-    name: Merge Coverage Artifacts
+  publish-coverage:
+    name: Publish Coverage to Pages
     needs: test
     if: >-
       github.ref == 'refs/heads/main' &&
       github.event_name == 'push' &&
       needs.test.outputs.passed == 'true'
-    runs-on: ubuntu-24.04
-    permissions:
-      actions: read
-    steps:
-      - name: Merge per-version coverage into single artifact
-        # yamllint disable-line rule:line-length
-        uses: actions/upload-artifact/merge@043fb46d1a93c77aae656e7c1c64a875d1fc6a0a # v7.0.1
-        with:
-          name: python-coverage
-          pattern: python-coverage-*
-          retention-days: 7
-
-  publish-coverage:
-    name: Publish Coverage to Pages
-    needs: [test, merge-coverage]
-    if: >-
-      github.ref == 'refs/heads/main' &&
-      github.event_name == 'push' &&
-      needs.test.outputs.passed == 'true'
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-test-python-publish.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-test-python-publish.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       coverage-percent: ${{ needs.test.outputs.coverage-percent }}
       upload-coverage: true
       job-name: Deploy Coverage to Pages
       runner-image: ubuntu-24.04
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
       egress-policy: block
       allowed-endpoints: >
         github.com:443

--- a/.github/workflows/test-ci.yml
+++ b/.github/workflows/test-ci.yml
@@ -49,6 +49,7 @@ jobs:
         astral.sh:443
         releases.astral.sh:443
     permissions:
+      actions: write  # aggregate job merges per-version python-coverage artifacts (v0.18.2+)
       contents: read
       pull-requests: write
 

--- a/.github/workflows/test-ci.yml
+++ b/.github/workflows/test-ci.yml
@@ -76,9 +76,29 @@ jobs:
             exit 1
           fi
 
+  merge-coverage:
+    name: Merge Coverage Artifacts
+    needs: test
+    if: >-
+      github.ref == 'refs/heads/main' &&
+      github.event_name == 'push' &&
+      needs.test.outputs.passed == 'true'
+    runs-on: ubuntu-24.04
+    permissions:
+      actions: read
+    steps:
+      - name: Merge per-version coverage into single artifact
+        # yamllint disable-line rule:line-length
+        uses: actions/upload-artifact/merge@043fb46d1a93c77aae656e7c1c64a875d1fc6a0a # v7.0.1
+        with:
+          name: python-coverage
+          pattern: python-coverage-*
+          delete-merged: true
+          retention-days: 7
+
   publish-coverage:
     name: Publish Coverage to Pages
-    needs: test
+    needs: [test, merge-coverage]
     if: >-
       github.ref == 'refs/heads/main' &&
       github.event_name == 'push' &&

--- a/.github/workflows/test-ci.yml
+++ b/.github/workflows/test-ci.yml
@@ -49,7 +49,8 @@ jobs:
         astral.sh:443
         releases.astral.sh:443
     permissions:
-      actions: write  # aggregate job merges per-version python-coverage artifacts (v0.18.2+)
+      # v0.18.2 aggregate job needs actions:write for artifact merge
+      actions: write
       contents: read
       pull-requests: write
 

--- a/.github/workflows/validate-action-pinning.yml
+++ b/.github/workflows/validate-action-pinning.yml
@@ -21,7 +21,7 @@ concurrency:
 jobs:
   validate:
     # yamllint disable-line rule:line-length
-    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-validate-action-pinning.yml@22a7c110415d75843c2c216f95ed47aaeb8e26fc
+    uses: lgtm-hq/lgtm-ci/.github/workflows/reusable-validate-action-pinning.yml@e42d374f1f89e0ad20d165cd611eb7732462b581
     with:
       egress-policy: block
       allowed-endpoints: >
@@ -30,6 +30,6 @@ jobs:
         codeload.github.com:443
         objects.githubusercontent.com:443
         raw.githubusercontent.com:443
-      tooling-ref: '22a7c110415d75843c2c216f95ed47aaeb8e26fc'
+      tooling-ref: 'e42d374f1f89e0ad20d165cd611eb7732462b581'
     permissions:
       contents: read

